### PR TITLE
frida-python: add aarch64-darwin support

### PR DESCRIPTION
**Pull request by stepanstipl**
## Description of changes

Add support for `aarch64-darwin` to `frida-python`.

Previously, the build would fail on unsupported platform, and subsequently on missing libraries when trying to build on MacOS with Arm/M1. This PR makes the build succeed on MacOS with M1. `frida-tools` package (from master/unstable), which depends on this and was previously blocked, was tested to be working as expected.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin

- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
  
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/frida-python/default.nix [hunk 1]
```diff
@@ -32,8 +43,17 @@ buildPythonPackage rec {
...
+    "-framework AppKit"
+  ]);
+
   propagatedBuildInputs = [ typing-extensions ];
 
+  buildInputs = [ ]
+  ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.AppKit
+  ];
+
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/frida-python/default.nix:56]: This is not really needed:

```suggestion
  buildInputs = lib.optionals stdenv.isDarwin [
    darwin.apple_sdk.frameworks.AppKit
  ];

```

**Review by AndersonTorres — COMMENTED** [review]
We do not use semantic commit messages.

Reword the commit as:

```
frida-python: enable support for aarch64-darwin
```

**Comment by stepanstipl** [comment]
Thank you @AndersonTorres for a good and constructive feedback, I believe I have now addressed all the points that you have raised. 👍 🙏

**Review by AndersonTorres — APPROVED** [review]

**Review by s1341 — APPROVED** [review]

File: pkgs/development/python-modules/frida-python/default.nix [hunk 2]
```diff
@@ -32,8 +43,16 @@ buildPythonPackage rec {
     popd
   '';
 
+  env.NIX_LDFLAGS = toString (lib.optionals stdenv.hostPlatform.isDarwin [
+    "-framework AppKit"
+  ]);
```
Comment (rhendric) [inline pkgs/development/python-modules/frida-python/default.nix:48]:
```suggestion
  env.NIX_LDFLAGS = lib.optionalString stdenv.hostPlatform.isDarwin "-framework AppKit";
```

**Comment by JohnRTitor** [comment]
@DontEatOreo could you test this?

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 315899` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>frida-tools</li>
    <li>frida-tools.dist</li>
    <li>python311Packages.frida-python</li>
    <li>python311Packages.frida-python.dist</li>
    <li>python312Packages.frida-python</li>
    <li>python312Packages.frida-python.dist</li>
  </ul>
</details>

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 315899` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Review by 4evy — APPROVED** [review]
LGTM!

**Pull request merged**